android/app/buildConfigs.gradle

```diff
@@ -0,0 +1,52 @@
+// See https://github.com/luggit/react-native-config/blob/master/android/dotenv.gradle for reference
+
+def createConfig () {
+    def buildConfig = [:]
+
+    if (!System.env["BUILD_CONFIG_NAME"]) {
+        println("*****************++++++++*********")
+        println("*** Missing BUILD_CONFIG_NAME ****")
+        println("************************++++++++**")
+
+        // Set to empty array to prevent failing for other tasks
+        project.ext.set("buildConfig", buildConfig)
+
+        return
+    }
+
+    def buildConfigName = System.env['BUILD_CONFIG_NAME']
+    println("Using build config $buildConfigName")
+
+    def command = "yarn babel-node tools/create-xcconfigs config $buildConfigName --directory android"
+    def proc = command.execute()
+    proc.waitFor()
+
+    if (proc.exitValue() != 0) {
+        println("Failed to create xcconfig file: $proc.err.text")
+        return
+    }
+    println("Created xcconfg file: $proc.in.text")
+
+
+    File f = new File("$project.rootDir/buildConfig.tmp.xcconfig");
+
+    f.eachLine { line ->
+        def parts = line.split(" = ")
+        buildConfig.put(parts[0], parts[1])
```

Comment:
Maybe some error checking would be nice here :)

Reply:
We also could use here a Java Properties file to load the xcconfig file: https://docs.oracle.com/javase/7/docs/api/java/util/Properties.html

Reply:
Good suggestion :+1: